File: ios/Examples/ObjectiveC/TrimExample.m
#import <VadCutIOS/VadCutIOS-Swift.h>

static TrimTask *TrimRecording(NSURL *inputURL, NSURL *outputURL) {
  VDTrimConfiguration *configuration = [VDTrimConfiguration new];
  configuration.minimumSilenceDurationMilliseconds = 700;
  return [VadCutObjC trimWithInputURL:inputURL
                            outputURL:outputURL
                         configuration:configuration
                               progress:^(NSInteger percent, NSString *phase) {
    NSLog(@"%@ %ld%%", phase, (long)percent);
  } completion:^(VDTrimResult *result, NSError *error) {
    if (error != nil) {
      NSLog(@"VadCut failed: %@", error);
    } else {
      NSLog(@"Output: %@", result.outputURL);
    }
  }];
}

static TrimTask *RemoveKnownRanges(NSURL *inputURL, NSURL *outputURL) {
  VDTrimConfiguration *configuration = [VDTrimConfiguration new];
  VDManualTrimPlan *plan = [VDManualTrimPlan removeRanges:@[
    [[VDAudioRange alloc] initWithStartMilliseconds:10000 endMilliseconds:15000],
    [[VDAudioRange alloc] initWithStartMilliseconds:42000 endMilliseconds:44500],
  ]];
  return [VadCutObjC trimWithInputURL:inputURL
                            outputURL:outputURL
                         configuration:configuration
                              manualPlan:plan
                               progress:nil
                             completion:^(VDTrimResult *result, NSError *error) {
    if (error != nil) {
      NSLog(@"VadCut failed (%@/%ld): %@", error.domain, (long)error.code, error);
    } else {
      NSLog(@"Removed %lld ms in %@", result.removedDurationMilliseconds,
            result.removedRanges);
    }
  }];
}

File: ios/Examples/Swift/TrimExample.swift
import Foundation
import VadCutIOS

func trimRecording(inputURL: URL, outputURL: URL) async throws -> TrimResult {
    try await VadCut.trim(
        TrimRequest(inputURL: inputURL, outputURL: outputURL, config: .preset(.voiceMemo))
    ) { progress in
        print("\(progress.phase.rawValue): \(progress.percent)%")
    }
}

func removeKnownRanges(inputURL: URL, outputURL: URL) async throws -> TrimResult {
    let manualPlan = ManualTrimPlan.removeRanges([
        AudioRange(startMilliseconds: 10_000, endMilliseconds: 15_000),
        AudioRange(startMilliseconds: 42_000, endMilliseconds: 44_500),
    ])
    return try await VadCut.trim(
        TrimRequest(
            inputURL: inputURL,
            outputURL: outputURL,
            manualTrimPlan: manualPlan
        )
    )
}

func keepKnownRanges(inputURL: URL, outputURL: URL) async throws -> TrimResult {
    try await VadCut.trim(
        TrimRequest(
            inputURL: inputURL,
            outputURL: outputURL,
            manualTrimPlan: .keepRanges([
                AudioRange(startMilliseconds: 0, endMilliseconds: 8_000),
                AudioRange(startMilliseconds: 12_000, endMilliseconds: 20_000),
            ])
        )
    )
}
